File: RiotGames/Dto/Match/InfoDto.cs
using Newtonsoft.Json;

namespace potg.RiotGames.Dto.Match;

public sealed record InfoDto
{
    [JsonProperty("gameCreation")]
    public ulong GameCreation { get; init; }

    [JsonProperty("gameDuration")]
    public int GameDuration { get; init; }

    [JsonProperty("gameEndTimestamp")]
    public ulong GameEndTimestamp { get; init; }

    [JsonProperty("gameId")]
    public long GameId { get; init; }

    [JsonProperty("gameMode")]
    public string GameMode { get; init; }

    [JsonProperty("gameName")]
    public string GameName { get; init; }

    [JsonProperty("gameStartTimestamp")]
    public ulong GameStartTimestamp { get; init; }

    [JsonProperty("gameType")]
    public string GameType { get; init; }

    [JsonProperty("gameVersion")]
    public string GameVersion { get; init; }

    [JsonProperty("mapId")]
    public int MapId { get; init; }

    [JsonProperty("participants")]
    public IReadOnlyCollection<ParticipantsDto> Participants { get; init; }

    [JsonProperty("platformId")]
    public string PlatformId { get; init; }

    [JsonProperty("queueId")]
    public int QueueId { get; init; }

    [JsonProperty("teams")]
    public IReadOnlyCollection<TeamDto> Teams { get; init; }

    [JsonProperty("tournamentCode")]
    public string TournamentCode { get; init; }
}

File: src/riot/lol/api/match/dto/info.dto.ts
import { Field, ObjectType } from "@nestjs/graphql";
import { ParticipantDto } from "./participant.dto";
import { TeamDto } from "./team.dto";

@ObjectType()
export class InfoDto {
  @Field()
  gameCreation: number;
  @Field()
  gameDuration: number;
  @Field()
  gameEndTimestamp: number;
  @Field()
  gameId: number;
  @Field()
  gameMode: string;
  @Field()
  gameName: string;
  @Field()
  gameStartTimestamp: number;
  @Field()
  gameType: string;
  @Field()
  gameVersion: string;
  @Field()
  mapId: number;
  @Field()
  participants: ParticipantDto;
  @Field()
  platformId: string;
  @Field()
  queueId: number;
  @Field()
  teams: TeamDto;
  @Field()
  tournamentCode: string;
}
